pulp_ansible/app/galaxy/v1/tasks.py

```diff
@@ -0,0 +1,184 @@
+        #     clone_url, github_reference, checkout_path=checkout_path
+        # )
+
+        # Upstream stores this as a "release date", pulp_ansible does not.
+        # github_commit_date = get_tag_commit_date(
+        #     clone_url, github_reference, checkout_path=checkout_path
+        # )
+
+        # pulp_ansible does not have role tags
+        """
+        # parse out the role metadata
+        role_meta = get_path_role_meta(checkout_path)
+        role_tags = role_meta.get("galaxy_info", {}).get("galaxy_tags", [])
+        if role_tags is None:
+            role_tags = []
+        """
+
+        # pulp_ansible does not have a model or a view for readme content
+        """
+        ldr = RoleLoader(
+            content_type="role",
+            root=os.path.dirname(checkout_path),
+            rel_path=os.path.basename(checkout_path),
+        )
+        readme = ldr._get_readme()
+        readme_html = markup_utils._render_from_markdown(readme)
+        """
+
+        metadata = {
+            "github_user": github_user,
+            "github_repo": github_repo,
+            "role_name": role_name,
+            "name": role_name,
+            "namespace": github_user,
+            "role_version": github_reference,
+        }
+
+        # pulp_ansible's Roles are actuallly Role Versions
+        # The roles list viewset aggregates the versions together
+        # by using a {namespace}.{name} as the primary key.
+        role = Role(
+            version=metadata["role_version"],
+            name=metadata["name"],
+            namespace=metadata["namespace"],
+        )
+        role.save()
+
+        relative_path = "%s/%s/%s.tar.gz" % (
+            metadata["github_user"],
+            metadata["role_name"],
+            metadata["role_version"],
+        )
+
+        # Make a tarball
+        shutil.rmtree(os.path.join(checkout_path, ".git"))
+        tarfn = f"/tmp/{github_user}.{role_name}.{github_reference}.tar.gz"
+        cmd = f"tar czvf {tarfn} {checkout_path}"
+        pid = subprocess.run(cmd, shell=True)
+        assert pid.returncode == 0
+
+        # Create a real artifact
+        artifact = Artifact.init_and_validate(tarfn)
+        artifact.save()
+
+        # This is a pulp'ism for the source url
+        # but we will generate the download_url in the
+        # serializer so this wouldn't ever be
+        # used ... ?
+        relative_path = "%s/%s/%s.tar.gz" % (
+            metadata["github_user"],
+            metadata["role_name"],
+            metadata["role_version"],
+        )
+
+        # Use Content to tie the role and artifact together
+        ca1 = ContentArtifact.objects.create(
+            artifact=artifact, content=role, relative_path=relative_path
+        )
+        ca1.save()
+
+        # Add the role to the legacy repository via a new version.
+        legacy = AnsibleRepository.objects.get(pulp_id=ansible_repo_id)
+        add_and_remove(int(legacy.pk), [role.pk], [])
```

Comment:
I'm surprised this works with the `int` if it does.

Reply:
Maybe because uuid is also just a disguised number.

```python
In [3]: int(uuid.uuid4())
Out[3]: 120903135329304400464259694501259332
```

But i agree it should not be casted here.

Reply:
fixed